src/models/predicate/predicate_nonexist.rs

```diff
@@ -0,0 +1,132 @@
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct PredicateNonexist {}
+
+impl From<LinkMetadataBuilder> for PredicateNonexist {
+    fn from(_meta: LinkMetadataBuilder) -> PredicateNonexist {
+        PredicateNonexist {}
+    }
+}
+
+impl PredicateLayout for PredicateNonexist {
+    fn to_bytes(&self) -> Result<Vec<u8>> {
+        Json::canonicalize(&Json::serialize(self)?)
+    }
+
+    fn into_enum(self: Box<Self>) -> PredicateWrapper {
+        PredicateWrapper::NonExist(*self)
+    }
+
+    fn version(&self) -> PredicateVersion {
+        PredicateVersion::NonExist
+    }
+
+    fn name(&self) -> Option<&String> {
+        None
+    }
+
+    fn materials(&self) -> Option<&BTreeMap<VirtualTargetPath, TargetDescription>> {
+        None
+    }
+
+    fn env(&self) -> Option<&BTreeMap<String, String>> {
+        None
+    }
+
+    fn byproducts(&self) -> Option<&BTreeMap<String, String>> {
+        None
+    }
+}
+
+#[cfg(test)]
+pub mod test {
+    use std::str;
+
+    use once_cell::sync::Lazy;
+
+    use super::PredicateNonexist;
+    use crate::models::{LinkMetadataBuilder, PredicateLayout, PredicateVersion, PredicateWrapper};
+
+    // TODO: move this to Metadata module after the revision of Metadata
+    pub static BLANK_META: Lazy<LinkMetadataBuilder> = Lazy::new(|| LinkMetadataBuilder::default());
+
+    pub static STR_NONENIST: &str = "{}";
+
+    pub static PREDICATE_NONENIST: Lazy<PredicateNonexist> = Lazy::new(|| PredicateNonexist {});
+
+    #[test]
+    fn into_trait_equal() {
+        let predicate = PredicateWrapper::NonExist(PREDICATE_NONENIST.clone());
+        let real = Box::new(PREDICATE_NONENIST.clone()).into_enum();
+
+        assert_eq!(predicate, real);
+    }
+
+    #[test]
+    fn create_predicate_from_meta() {
+        let predicate = PredicateWrapper::from_meta(BLANK_META.clone(), PredicateVersion::NonExist);
+        let real = Box::new(PREDICATE_NONENIST.clone()).into_enum();
+
+        assert_eq!(predicate, real);
+    }
+
+    #[test]
+    fn serialize_predicate() {
+        let predicate = Box::new(PREDICATE_NONENIST.clone()).into_enum();
+        let buf = predicate.into_trait().to_bytes().unwrap();
+        let predicate_serialized = str::from_utf8(&buf).unwrap();
+
+        assert_eq!(predicate_serialized, STR_NONENIST);
+    }
+
+    #[test]
+    fn deserialize_predicate() {
+        let predicate = PredicateWrapper::from_bytes(
+            STR_NONENIST.as_bytes().to_vec(),
+            PredicateVersion::NonExist,
+        )
+        .unwrap();
+        let real = Box::new(PREDICATE_NONENIST.clone()).into_enum();
+
+        assert_eq!(predicate, real);
+    }
+
+    #[test]
+    fn deserialize_auto() {
+        let predicate = PredicateWrapper::try_from_bytes(STR_NONENIST.as_bytes().to_vec()).unwrap();
+        let real = Box::new(PREDICATE_NONENIST.clone()).into_enum();
+
+        assert_eq!(predicate, real);
+    }
+
+    // #[test]
```

Comment:
Why is this commented out?

Reply:
This test relies on predicate `V02`, and it ihas been uncommented now.